src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -107,48 +100,60 @@ export default class PhotoField extends React.Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const newView = this.state.src && (prevState.src !== this.state.src || (prevState.done && !this.state.done));
+    const newView = this.state.src && (prevState.src !== this.state.src || ((this.state.progress === 100) && (this.state.progress < 100)));
     const cropper = this.refs.cropper;
     if (cropper && newView) {
       this.setCropBox();
     }
   }
 
   onEdit = () => {
-    this.setState({ done: false, warningMessage: null });
+    this.setState({ progress: 0, warningMessage: null });
   }
 
   onDone = () => {
-    this.setState({ done: true, warningMessage: null });
+    const filePath = 'data.photo';
+    this.props.formContext.uploadFile(
```

Comment:
We don't know quite yet if we'll need to call this. Ryan B's working with Patrick to do load testing this week. If we can store images in the database as base64, then I think we'll want to remove this and call `onChange` with an object that's formatted like the other file fields. If not, we would keep this.

Reply:
Thanks for the helpful context.